go/integration_test/processor/processor_gas_test.go

```diff
@@ -0,0 +1,503 @@
+				Value:     tosca.NewValue(3),
+				Nonce:     4,
+			},
+			After: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 4},
+			},
+			Receipt: tosca.Receipt{
+				Success: false,
+				GasUsed: 0,
+			},
+			OperaError: fmt.Errorf("gas too low"),
+		},
+		"SimpleCodeExact": {
+			Before: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 4},
+				{2}: Account{Balance: tosca.NewValue(0),
+					Code: tosca.Code{
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.RETURN),
+					},
+				},
+			},
+			Transaction: tosca.Transaction{
+				Sender:    tosca.Address{1},
+				Recipient: &tosca.Address{2},
+				GasLimit:  floria.TxGas + codePrice,
+				Nonce:     4,
+			},
+			After: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 5},
+				{2}: Account{Balance: tosca.NewValue(0),
+					Code: tosca.Code{
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.RETURN),
+					},
+				},
+			},
+			Receipt: tosca.Receipt{
+				Success: true,
+				GasUsed: floria.TxGas + codePrice,
+			},
+		},
+		"SimpleCodeSufficient": {
+			Before: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 4},
+				{2}: Account{Balance: tosca.NewValue(0),
+					Code: tosca.Code{
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.RETURN),
+					},
+				},
+			},
+			Transaction: tosca.Transaction{
+				Sender:    tosca.Address{1},
+				Recipient: &tosca.Address{2},
+				GasLimit:  floria.TxGas + codePrice + excessGas,
+				Nonce:     4,
+			},
+			After: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 5},
+				{2}: Account{Balance: tosca.NewValue(0),
+					Code: tosca.Code{
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.RETURN),
+					},
+				},
+			},
+			Receipt: tosca.Receipt{
+				Success: true,
+				GasUsed: floria.TxGas + codePrice + excessGas/10,
+			},
+		},
+		"SimpleCodeInsufficient": {
+			Before: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 4},
+				{2}: Account{Balance: tosca.NewValue(0),
+					Code: tosca.Code{
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.RETURN),
+					},
+				},
+			},
+			Transaction: tosca.Transaction{
+				Sender:    tosca.Address{1},
+				Recipient: &tosca.Address{2},
+				GasLimit:  floria.TxGas + codePrice - 1,
+				Nonce:     4,
+			},
+			After: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 5},
+				{2}: Account{Balance: tosca.NewValue(0),
+					Code: tosca.Code{
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.RETURN),
+					},
+				},
+			},
+			Receipt: tosca.Receipt{
+				Success: false,
+				GasUsed: floria.TxGas + codePrice - 1,
+			},
+			OperaError: fmt.Errorf("gas too low"),
```

Comment:
this error is used multiple times, maybe it's worth predeclaring as a variable or const.

Reply:
The code has been refactored to only use this error in one place.